protocol/streaming/full_node_streaming_manager_test.go

```diff
@@ -0,0 +1,476 @@
+				TotalFilledQuantums: totalFilledQuantums,
+			},
+		},
+	}
+}
+
+func toStreamUpdate(offChainUpdates ...ocutypes.OffChainUpdateV1) clobtypes.StreamUpdate {
+	return clobtypes.StreamUpdate{
+		BlockHeight: uint32(0),
+		ExecMode:    uint32(sdktypes.ExecModeFinalize),
+		UpdateMessage: &clobtypes.StreamUpdate_OrderbookUpdate{
+			OrderbookUpdate: &clobtypes.StreamOrderbookUpdate{
+				Updates:  offChainUpdates,
+				Snapshot: true,
+			},
+		},
+	}
+}
+
+type MockMessageSender struct{}
+
+func (mms *MockMessageSender) Send(*clobtypes.StreamOrderbookUpdatesResponse) error {
+	return nil
+}
+
+func NewOrderbookSubscription(
+	ids []uint32,
+	updatesChannel chan []clobtypes.StreamUpdate,
+) *streaming.OrderbookSubscription {
+	sIds := []satypes.SubaccountId{}
+	for _, id := range ids {
+		sIds = append(sIds, satypes.SubaccountId{Owner: owner, Number: id})
+	}
+	return streaming.NewOrderbookSubscription(
+		0,
+		[]uint32{},
+		sIds,
+		[]uint32{},
+		&MockMessageSender{},
+		updatesChannel,
+	)
+}
+
+func NewStreamOrderbookFill(
+	blockHeight uint32,
+	execMode uint32,
+) *clobtypes.StreamUpdate {
+	return &clobtypes.StreamUpdate{
+		BlockHeight: blockHeight,
+		ExecMode:    execMode,
+		UpdateMessage: &clobtypes.StreamUpdate_OrderFill{
+			OrderFill: nil,
+		},
+	}
+}
+
+func NewStreamTakerOrder(
+	blockHeight uint32,
+	execMode uint32,
+	order *clobtypes.Order,
+	remainingQuantums uint64,
+	optimisticallyFilledQuantums uint64,
+) *clobtypes.StreamUpdate {
+	return &clobtypes.StreamUpdate{
+		BlockHeight: blockHeight,
+		ExecMode:    execMode,
+		UpdateMessage: &clobtypes.StreamUpdate_TakerOrder{
+			TakerOrder: &clobtypes.StreamTakerOrder{
+				TakerOrder: &clobtypes.StreamTakerOrder_Order{
+					Order: order,
+				},
+				TakerOrderStatus: &clobtypes.StreamTakerOrderStatus{
+					OrderStatus:                  uint32(clobtypes.Success),
+					RemainingQuantums:            remainingQuantums,
+					OptimisticallyFilledQuantums: optimisticallyFilledQuantums,
+				},
+			},
+		},
+	}
+}
+
+func NewSubaccountUpdate(
+	blockHeight uint32,
+	execMode uint32,
+	subaccountId *satypes.SubaccountId,
+) *clobtypes.StreamUpdate {
+	return &clobtypes.StreamUpdate{
+		BlockHeight: blockHeight,
+		ExecMode:    execMode,
+		UpdateMessage: &clobtypes.StreamUpdate_SubaccountUpdate{
+			SubaccountUpdate: &satypes.StreamSubaccountUpdate{
+				SubaccountId:              subaccountId,
+				UpdatedPerpetualPositions: []*satypes.SubaccountPerpetualPosition{},
+				UpdatedAssetPositions:     []*satypes.SubaccountAssetPosition{},
+				Snapshot:                  true,
+			},
+		},
+	}
+}
+
+func NewPriceUpdate(
+	blockHeight uint32,
+	execMode uint32,
+) *clobtypes.StreamUpdate {
+	return &clobtypes.StreamUpdate{
+		BlockHeight: blockHeight,
+		ExecMode:    execMode,
+		UpdateMessage: &clobtypes.StreamUpdate_PriceUpdate{
+			PriceUpdate: &pricestypes.StreamPriceUpdate{
+				MarketId: 1,
+				Price: pricestypes.MarketPrice{
+					Id:       1,
+					Exponent: 1_000_000,
```

Comment:
Nit: This proto represents a price of `Price * 10 ^ Exponent` so a more realistic value would be `Exponent = 6`